src/services/user.services.ts

```diff
@@ -0,0 +1,23 @@
+import { AuthenticationError } from 'apollo-server';
+import { User } from '../sequelize/models/user.models';
+import { unHashPassword } from '../utils/user.utils';
+import { IloginData } from '../interfaces/user.iterfaces';
+
+
+class UserService {
+  static async loginUser(user: IloginData) {
+    const { email, password } = user;
+    const findUser = await this.findUserByEmail(email);
+    if (!findUser || !unHashPassword(password, findUser.password)) {
+      throw new AuthenticationError('Invalid email or password');
```

Comment:
@harerakalex Wouldnt this result in some uncaught exception since there is not try catch block here or in the resolvers. winston will report this but we should strive to handle errors in a more predictable manner

Reply:
That's a good idea but since graphQL is able to handle any possible errors which may arise. In my opinion, I think it wouldn't be better to use try/catch to handle the errors.

Reply:
okay